packages/styled-components/src/sheet/test/GroupedTag.test.js

```diff
@@ -89,7 +89,7 @@ it('does supports large group numbers', () => {
   expect(groupedTag.length).toBeGreaterThan(group);
   expect(tag.length).toBe(1);
   expect(groupedTag.indexOfGroup(group)).toBe(0);
-  expect(groupedTag.getGroup(group)).toBe('.test {}\n');
+  expect(groupedTag.getGroup(group)).toBe('.test {}/*!sc*/\n');
```

Comment:
I don't understand why this change adds this string to the output of this method. How do we know this change isn't introducing a new, unwanted behavior, if all these tests needed to change?

Reply:
This is used by `outputSheet` in the SSR functionality. We need to add the marker at this point so the individual rules are split correctly during rehydration. This PR generally changes the whole rehydration mechanic, which previously used regexes to do the splitting.